internal/model/recommendation_set.go

```diff
@@ -48,13 +47,12 @@ func (r *RecommendationSet) GetRecommendationSets(orgID string, orderQuery strin
 			JOIN rh_accounts ON clusters.tenant_id = rh_accounts.id
 		`).Model(r).Preload("Workload.Cluster.RHAccount").Where("rh_accounts.org_id = ?", orgID)
 
```

Comment:
Above query is returning incorrect results. We need to select `container_name` from recommendation_set table make a join on it. 
request you to change above query with the below.
```
query := db.Table("recommendation_sets").Joins(`
		JOIN (
			SELECT workload_id, container_name, MAX(monitoring_end_time) AS latest_monitoring_end_time 
			FROM recommendation_sets GROUP BY workload_id, container_name
		) latest_rs ON recommendation_sets.workload_id = latest_rs.workload_id 
				AND recommendation_sets.monitoring_end_time = latest_rs.latest_monitoring_end_time
				AND recommendation_sets.container_name = latest_rs.container_name
			JOIN workloads ON recommendation_sets.workload_id = workloads.id
			JOIN clusters ON workloads.cluster_id = clusters.id
			JOIN rh_accounts ON clusters.tenant_id = rh_accounts.id
		`).Model(r).Preload("Workload.Cluster.RHAccount").Where("rh_accounts.org_id = ?", orgID)
```

Reply:
@patilsuraj767 Could you let me know about the issue in brief?

Reply:
Changes pushed